### ISSUE TITLE
[BUG] Multiple consecutive AND conditions not fully translated to LLVM IR

### ISSUE THREAD

**Issue by huajie-zhong** [cornell-zhang/allo#355]
**Describe the bug**
When translating Python code with multiple consecutive logical AND conditions into LLVM IR, only the first set of conditions is being checked while subsequent conditions are ignored. This creates a serious memory safety vulnerability as bound checking is incomplete.

**To Reproduce**
The issue occurred when attempting to implement conv2d via allo.

Issue at `if h_pos >= 0 and h_pos < H and w_pos >= 0 and w_pos < W:`
The statement work if change to explicit condition split: `if (h_pos >= 0 and h_pos < H) and (w_pos >= 0 and w_pos < W):`

For variable meanings:
`Ty`: Number Type
`B`: batch size
`Cin/Cout`: Channel in/out
`H/W`: height/width
`Kh/Kw`: kernal height/width
`Oh/Ow`: Out height/width
`Pd0/Pd1`: Padding first/second dimension

`inp`: input
`filter`: the filter/kernal

```
def conv2d[Ty, B, Cin, Cout, H, W, Kh, Kw, Oh, Ow, Pd0, Pd1](
    inp: "Ty[B, Cin, H, W]", filter: "Ty[Cout, Cin, Kh, Kw]", bias: "Ty[Cout]"
) -> "Ty[B, Cout, Oh, Ow]":
    # https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html
    Z: Ty[B, Cout, Oh, Ow]
    
    # Current implementation is does not support dilation and stride other than 1
    for batch in range(B):
        for cout in range(Cout):
            for oh in range(Oh):
                for ow in range(Ow):
                    sum: Ty = bias[cout]

                    for cin in range(Cin):
                        for kh in range(Kh):
                            for kw in range(Kw):
                                h_pos: Ty = oh + kh - Pd0
                                w_pos: Ty = ow + kw - Pd1
                                if h_pos >= 0 and h_pos < H and w_pos >= 0 and w_pos < W:
                                    sum += (
                                        inp[batch, cin, h_pos, w_pos]
                                        * filter[cout, cin, kh, kw]
                                    )

                    Z[batch, cout, oh, ow] = sum
    return Z
```

**Buggy output**
Without explicit parenthesis,  the buggy LLVM IR only includes bounds checking for the first variable (h_pos):

```
%139 = llvm.fcmp "oge" %138, %31 : f32    
%140 = llvm.fcmp "olt" %138, %30 : f32    
%141 = llvm.and %139, %140  : i1          
llvm.cond_br %141, ^bb15, ^bb16 
```

**Expected behavior**
The expected should be:
```
%139 = llvm.fcmp "oge" %138, %31 : f32
%140 = llvm.fcmp "olt" %138, %30 : f32
%141 = llvm.and %139, %140  : i1

%144 = llvm.fcmp "oge" %143, %31 : f32
%145 = llvm.fcmp "olt" %143, %30 : f32
%146 = llvm.and %144, %145  : i1

%147 = llvm.and %141, %146  : i1
llvm.cond_br %147, ^bb15, ^bb16
```

**Additional context**
This seems to be hard to reproduce via simple example. I encountered this issue and wished to create a smaller example, but they seemed to work for my small examples.

I've also attached the full MLIR modules for both correct and incorrect output below

[correct.txt](https://github.com/user-attachments/files/19985962/correct.txt)
[incorrect.txt](https://github.com/user-attachments/files/19985963/incorrect.txt)
